Formalize language and improve grammar in "Shading Language" (#4048)
godotengine/godot-docs: tutorials/shading/shading_reference/shading_language.rst

This is rarely needed in the vertex processor function (where full precision is needed most of the time),
but is often useful in the fragment processor.

> Some architectures (mainly mobile) can benefit greatly from this, but may have downsides

Reviewer: Instead say "but there are downsides".

> Some architectures (mainly mobile) can benefit greatly from this, but may have downsides
> such as additional overhead of conversion between precisions. Refer to the documentation of the 
> target architecture for further information. In many cases, mobile drivers cause inconsistent
> or unexpected behaviour and it is best to avoid specifying precision unless necessary.

Reviewer 2: ```suggestion
Some architectures (mainly mobile) can benefit significantly from this, but there are downsides such
as the additional overhead of conversion between precisions. Refer to the documentation of the
target architecture for further information. In many cases, mobile drivers cause inconsistent or
unexpected behavior and it is best to avoid specifying precision unless necessary.
```
